### ISSUE TITLE
RouteNotFoundException: Route "farm.quick.drilling" does not exist.

### ISSUE THREAD

**Issue by mstenta** [Rothamsted-Ecoinformatics/farm_rothamsted#570]
I started seeing these error notifications this morning:

```
Symfony\\Component\\Routing\\Exception\\RouteNotFoundException: Route \u0022farm.quick.drilling\u0022 does not exist. in Drupal\\Core\\Routing\\RouteProvider-\u003EgetRouteByName() (line 206 of \/opt\/drupal\/web\/core\/lib\/Drupal\/Core\/Routing\/RouteProvider.php).
```

This must be related to the recent farmOS core v2.2.0 upgrade that took place on Saturday.

**Comment by mstenta** [comment]
I put a hotfix into place on the Rothamsted instances, and it seems to be working again. I will describe what I found, what I did, and what we need to do next in more detail in the next comment...

**Comment by mstenta** [comment]
Post-mortem:

- farmOS v2.2.0 included this PR to enable "Configurable Quick Forms": https://github.com/farmOS/farmOS/pull/707
- One of the commits in that PR merged all quick form routes together into a single `farm_quick.quick_form` route: https://github.com/farmOS/farmOS/pull/707/commits/3505e80124a2fe365eab5510fd2585f9ead9e69d
- This change broke quick form actions, each of which point to a specific route via their `confirm_form_route_name` annotation.
- There isn't a simple fix for that because Drupal core doesn't seem to support passing additional parameters to `confirm_form_route_name` routes, so there is no way to refactor quick form actions so that they pass the `quick_form` route parameter (afaict): https://git.drupalcode.org/project/drupal/-/blob/df5948229a8dc10bef675acb31ee939037b863ca/core/modules/views/src/Plugin/views/field/BulkForm.php#L431
- So, the quickest way to fix it was to revert https://github.com/farmOS/farmOS/pull/707/commits/3505e80124a2fe365eab5510fd2585f9ead9e69d
- Since that is a farmOS core change, it will require its own PR, review, and a new release, which would mean Rothamsted quick form actions would be broken in the meantime.
- Rather than wait for that, I reverted the commit locally (resolving conflicts in the process) and created a `hotfix.patch` (attached), which I deployed to the Rothamsted instances on Farmier specifically. Clearing the caches after applying that patch seems to have fixed the issue.

This is a *temporary* fix. Ultimately we need to fix this in farmOS core. I will discuss with @paul121 and we'll figure out the permanent fix. I will leave this issue open in the meantime...

[hotfix.patch.txt](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/files/12846509/hotfix.patch.txt)

**Comment by mstenta** [comment]
I opened an upstream bug report: https://github.com/farmOS/farmOS/issues/727

**Comment by mstenta** [comment]
Fixed this upstream and will tag a 2.2.1 release of farmOS shortly. Closing this.
